### ISSUE TITLE
Remove the auto initializing behavior

### ISSUE THREAD

**Issue by Windvis** [redpencilio/ember-plausible#4]
At the moment it's possible to start and configure the Plausible service from the config/environment file. We read this config in an initializer and start the service when needed. I'm not much of a fan of this flow and only added it since that's how it was traditionally done in addons.

We should drop support for the config object in config/environment and require that users use the `enable` method on the service directly. This would allow us to remove the custom `"true"` handling logic which is only needed because of the string replacement environment variable system we use in our docker images. Apps can simply run the `enable` method conditionally based on the environment value.

Before:
```diff
// config/environment.js
module.exports = function (environment) {
  let ENV = {
    // ...
-    'ember-plausible': {
-    enabled: '{{PLAUSIBLE_ENABLED}}', // This will be replaced with "true" if a certain environment wants it enabled
-    // other plausible config
-    },
+   plausibleEnabled: '{{PLAUSIBLE_ENABLED}}', // This will be replaced with "true" if a certain environment wants it enabled
  };
};
```

After:
```js
// app/routes/application.js
...
beforeModel() {
  if (config.plausibleEnabled) {
    this.plausible.enable({
      // other plausible config
    })
  }
}
```

It's a bit more code, but it will make the addon less mu-semtech / Semantic.Works specific.

**Comment by nvdk** [comment]
I think allowing the host app when to enable tracking is a very useful feature, but I don't really see the added value in having to pass in the config. might be missing something :)

**Comment by Windvis** [comment]
TL;DR no major benefits but it simplifies the addon code and is more in line with my personal preferences.

> I still prefer to keep the automatic startup behaviour. It would be a "set-up and forget about it" feature. If you need to procedurally start analytics, that code is buried in places I would not think to look at.

I think that's just personal preference. If I want to know why something (runtime code) started globally I would look at the application route hooks, or the initializers folder. I prefer the more explicit enabling over something that's magically done behind the scenes based on some configuration options. I don't see how adding some config makes something more "set and forget" over adding some code either. It's potentially less boilerplate, but I don't necessarily see that as a benefit.

> Manually enabling and disabling analytics is always an option through the service, for apps that do need precise control.
Or am I completely missing the ball?

My main motivator for this was the fact that the addon includes Semantic Works specific code. [This](https://github.com/redpencilio/ember-plausible/blob/d9dc25c4440f8b3ab5aaf19abe4126104a63205a/addon/instance-initializers/plausible.js#L11-L18) code is only there because we needed to support the env string replacement that we do in our apps. I would really like to get rid of it.

It also means we don't have to add FastBoot specific code to the addon. The user can simply call `enable` when needed. Sure, that's already possible at the moment, but it still requires you to explicitly disable the auto startup behavior.

So if we remove the Semantic Works specific code, and require `config['ember-plausible'].enabled` to be a boolean, all our projects will need the manual `enable` call anyways, so we might as well make it the only option and remove the initializer altogether.

> Also, isn't beforeModel() on the application route executed on every route transition?

You never leave the application route so it doesn't run on every transition. This does make me realize that using something like the new [`RouterService#refresh` method](https://api.emberjs.com/ember/4.4/classes/RouterService/methods/refresh?anchor=refresh) without a pivot route _will_ refresh those hooks. So users would have to set a flag on the route, or move the logic to initializers. This is something that we need to check regardless because we (and the Ember community as a whole) frequently use that hook for other setup related code already.
